datadog_checks_base/datadog_checks/base/utils/agent/memory.py

```diff
@@ -33,6 +33,14 @@
 TRACE_LOCK = threading.Lock()
 
 
+class MemoryProfileMetric(object):
+    __slots__ = ('name', 'value')
+
+    def __init__(self, name, value):
+        self.name = 'datadog.agent.profile.memory.{}'.format(name)
```

Comment:
perhaps we could put the `datadog.agent.profile.memory` namespace in a global, or at least `datadog.agent.profile` so we can reuse when/if the `cpu` profiling metrics come along?